### ISSUE TITLE
ENH: Expose dropna keyword in dissolve (Null values disappear)

### ISSUE THREAD

**Issue by philippe-code** [geopandas/geopandas#1814]
## CODE
```py
import geopandas

shp = r"C:\Users\LEBPH5\Desktop\carte_a_jour\Donnees_test\histo_majf.shp"
outfc = r"C:\Users\LEBPH5\Desktop\carte_a_jour\Donnees_test\HISTO_MAJF_2018_Dissolved.shp"
fields = 'ORIGINE'

layer = geopandas.read_file(shp)
dissolved = layer.dissolve(by=fields)
dissolved.to_file(outfc, driver="ESRI Shapefile")
```

## PROBLEM
When I try to dissolve a shapefile by a field with some NULL values, the NULL values disappear in the result shapefile. 

## QUESTION

I want to keep the Null values in the result file and I want the Null values to be dissolved.
I found a method that work. I can calculate a string (example 'NA') for the Null values and after dissolve my shapefile. And after the dissolve, recalculate the value 'NA' to Null but it can be very long because I have large dataset to treat.

It's there a way to do it directly with geopandas? thank you

## IMAGES

![hist_maj_table](https://user-images.githubusercontent.com/59576399/106777357-ec030f00-6612-11eb-9ea9-3318abe2ea0f.JPG)
![HISTO_MAJF_2018_Dissolved_Table](https://user-images.githubusercontent.com/59576399/106777360-ec9ba580-6612-11eb-9fca-01048f6d1b56.JPG)
![Missing_null](https://user-images.githubusercontent.com/59576399/106777361-ec9ba580-6612-11eb-906e-e11079e8f4ee.JPG)

**Comment by martinfleis** [comment]
Hmm, pandas as `dropna` keyword for this situation. We should expose it as well in `dissolve` to have control over this situation.

For now, you can replace missing values with a string or something before dissolving. 

```py
import geopandas as gpd

gdf = gpd.read_file(gpd.datasets.get_path('naturalearth_lowres'))
gdf_none = gdf.replace({"Europe": None})  # this introduces missing values

filled = gdf_none.fillna('null')  # replacing missing values with `'null'` which is a string in this case
dissolved = filled.dissolve('continent')
```

**Comment by johnflavin** [comment]
@martinfleis This can be closed after #1845.
